include/nighthawk/common/milestone_tracker.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+#include "envoy/common/pure.h"
+#include "envoy/common/time.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Nighthawk {
+
+class MilestoneTracker {
```

Comment:
What this models is very similar to a trace span in something like Zipkin or OpenCensus. Envoy has some abstractions for this, see https://github.com/envoyproxy/envoy/blob/master/include/envoy/tracing/http_tracer.h#L69. I wonder if there is any useful overlap here?

Should we just use some tracing library like OpenCensus? Is the overhead and resolution of these sufficient for tracking time inside NH? This also reminds me a bit of VMware Kstats.